swarm/storage/resource.go

```diff
@@ -868,16 +822,24 @@ func isMultihash(data []byte) int {
 // TODO: this should not be exposed, but swarm/testutil/http.go needs it
 func NewTestResourceHandler(datadir string, ethClient headerGetter, validator ResourceValidator) (*ResourceHandler, error) {
 	path := filepath.Join(datadir, DbDirName)
-	basekey := make([]byte, 32)
 	hasher := MakeHashFunc(SHA3Hash)
-	dbStore, err := NewLDBStore(path, hasher, singletonSwarmDbCapacity, func(k Key) (ret uint8) { return uint8(Proximity(basekey[:], k[:])) })
+	params := NewLDBStoreParams(path, 0, nil, nil)
+	params.Validator = func(hash SwarmHash, key *Key, data []byte) bool {
+		log.Trace("skipping check on key in resource update test")
+		return true
+	}
+	dbStore, err := NewLDBStore(params)
 	if err != nil {
 		return nil, err
 	}
 	localStore := &LocalStore{
-		memStore: NewMemStore(dbStore, singletonSwarmDbCapacity),
+		memStore: NewMemStore(NewStoreParams(defaultDbCapacity, nil, nil), dbStore),
 		DbStore:  dbStore,
 	}
-	resourceChunkStore := NewResourceChunkStore(localStore, nil)
+	resourceChunkStore := localStore //NewResourceChunkStore(localStore, nil)
 	return NewResourceHandler(hasher, resourceChunkStore, ethClient, validator)
 }
+
+func ValidateResourceChunk(hash SwarmHash, key *Key, data []byte) bool {
```

Comment:
I thought this was already implemented

Reply:
@zelig Yes, but this is the glue needed to use it from the chunk stores.